### PR TITLE
feat(update): Add update command

### PR DESCRIPTION
**Pull request by PWKad**
This adds the ability to update official aurelia dependencies in one command

### CODE REVIEW

**Comment by PWKad** [comment]
@joelcoxokc @zewa666 @ahmedshuhel The only issue remaining that I see is jspm clean is run too early once the first promise is resolved.  Gladly accept any ideas for fixing that otherwise we can open an issue to add that as an enhancement to spawn-promise which should resolve that issue.

File: src/commands/updater.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import * as logger from '../lib/logger';
```
Comment (zewa666) [inline src/commands/updater.js]: So according to our last discussion should we do this one via `require`?

File: src/lib/updater/index.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  var updateCommands = [];
+  var repoList = readConfigJs();
+  unbundle().then(function(){
+    installer.runNPMInstall(function(){
+      logger.log('Successfully npm installed');
+      logger.log('Updating all aurelia libs');
+      repoList.forEach(repo => {
+        updateCommands.push({ command: 'jspm', args: ['install', repo]})
+      });
+      SpawnPromise(updateCommands)
```
Comment (zewa666) [inline src/lib/updater/index.js]: Guess this is what you meant with the synchronous multi execution. Guess we should, as mentioned in the gitter channel, use something like Bluebirds Promise.join feature

File: src/lib/updater/index.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  });
+}
+
+function moveJSPMPackages(done){
+  fs.rename('./jspm_packages', './jspm_packages_backup', done);
+}
+
+function moveNPMPackages(done){
+  fs.rename('node_modules', 'node_modules_backup', done);
+}
```
Comment (zewa666) [inline src/lib/updater/index.js]: If you'd like to rewrite those to promises you can use bluebirds promisifyAll feature. https://github.com/petkaantonov/bluebird/blob/master/API.md#promisepromisifyallobject-target--object-options---object

File: src/lib/updater/index.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+          for (var p in v) {
+            if (!v.hasOwnProperty(p))
+              continue;
+            cfg[c][p] = v[p];
+          }
+        }
+        else
+          cfg[c] = v;
+      }
+    },
```
Comment (zewa666) [inline src/lib/updater/index.js]: Phew that one's really hard to read. Now this is only my opinion so feel free to ignore if you don't like it but I'd add braces even on single step if's as it makes the code more readable. Next the variable naming convention makes understanding this section quite hard, so longer more descriptive names could help.

File: src/start.js [hunk 5]
```diff
@@ -127,4 +127,26 @@ export function start() {
...
+          , info : 'Blow it up'
+          , required: false
+        },
+      });
+
+      // logger.log('  Examples:');
+      // logger.log();
+      // logger.log('    $ aurelia plugin -a i18n');
+      // logger.log('    $ aurelia p -r i18n -c');
+      // logger.log();
```
Comment (zewa666) [inline src/start.js]: Are those comments meant as an example output of the --help method above?

File: src/lib/updater/index.js (hunk 4 above)
Comment (PWKad) [inline src/lib/updater/index.js]: Yeah most of this is borrowed from [jspm config loader](https://github.com/jspm/jspm-cli/blob/master/lib/config/loader.js#L68-L92) for now until I can find a way to take advantage of that loader through their api.

File: src/start.js (hunk 5 above)
Comment (PWKad) [inline src/start.js]: Good question, I had that there uncommented before but I noticed that all of the examples were commented above in the same file.  Actually while reading it those are from the plugin anyway so need to remove.

Comment (joelcoxokc) [inline src/start.js]: I have actually not been adding braces around simple conditions... I think braces are only necessary if the condition block is large.

for example I wouldn't use braces here

```javascript
if (1 === num)
    return 'yes';
```

Comment (joelcoxokc) [inline src/start.js]: I like this PR. I would like to merge. 

Or would you rather wait for my PR with the updated spawnPromise?

**Comment by joelcoxokc** [comment]
@PWKad Would you like me to convert this PR to our new API?

**Pull request closed without merging**